crates/test-utils/src/authority.rs

```diff
@@ -70,3 +84,80 @@ pub fn create_authority_aggregator(
         .collect();
     AuthorityAggregator::new(committee, clients)
 }
+
+/// Get a network client to communicate with the consensus.
+pub fn get_client(config: &ValidatorInfo) -> NetworkAuthorityClient {
+    NetworkAuthorityClient::connect_lazy(config.network_address()).unwrap()
+}
+
+/// Submit a certificate containing only owned-objects to all authorities.
+pub async fn submit_single_owner_transaction(
+    transaction: Transaction,
+    configs: &[ValidatorInfo],
+) -> Vec<TransactionInfoResponse> {
+    let certificate = make_certificates(vec![transaction]).pop().unwrap();
+    let txn = ConfirmationTransaction { certificate };
+
+    let mut responses = Vec::new();
+    for config in configs {
+        let client = get_client(config);
+        let reply = client
+            .handle_confirmation_transaction(txn.clone())
+            .await
+            .unwrap();
+        responses.push(reply);
+    }
+    responses
+}
+
+/// Keep submitting the certificates of a shared-object transaction until it is sequenced by
+/// at least one consensus node. We use the loop since some consensus protocols (like Tusk)
+/// may drop transactions. The certificate is submitted to every Sui authority.
```

Comment:
But is the resubmission not usually done by the validator?

Reply:
Yes it is. These facilities are just test clients